tests/legacy/query_model_convert_to_new.py

```diff
@@ -127,9 +130,10 @@ def convert_value_from_aggregate(node: old.ValueFromAggregate):
     old_aggregate = node.source
     assert isinstance(old_aggregate, old.RowFromAggregate)
     source = convert_node(old_aggregate.source)
-    column = select_column(source, old_aggregate.input_column)
     Aggregation = AGGREGATE_MAP[old_aggregate.function]
-    return Aggregation(column)
+    if Aggregation not in AGGREGATIONS_ON_FRAMES:
+        source = select_column(source, old_aggregate.input_column)
```

Comment:
Perhaps this answers my question above -- the immediate source of an aggregation must be a `SelectColumn` because that's all that the old query model supports. But (I think) the new query model supports an arbitrary source and we'll get weird errors if someone tries it. Perhaps we need an assert with a clear message in `query_model_convert_to_old` for when we hit that case.

Reply:
Yes, the two conversion modules are almost completely uncommented and I didn't make much effort to make them particularly readable either. This was, of course, a calculated decision to encourage us to get rid of them as soon as possible :stuck_out_tongue: 

My approach with the `convert_to_old` module was just to exactly invert the translation done by `convert_to_new` and assert that the constructs are in the specific form that that translation generates. I didn't spend any time thinking about whether the existing query engine could support a wider range of queries.

The state I was aiming for, and which I think we've now achieved with the validation code, is:
 * If you can construct a query with the query model then you've constructed something meaningful.
 * If you get an assertion error from the translation code, or some other error from the query engine, then that means we've hit a limitation of the existing query engine and we either need to do some refactoring work or decide that we're not  yet supporting that construction.
 * If the query executes without error then the results should be correct.

I think all this does make clear though that we need to start the refactoring work sooner rather than later.

Reply:
Yeah, fair enough.